pkg/controller/apis/config/fuzzer/fuzzer.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fuzzer
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/gofuzz"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeserializer "k8s.io/apimachinery/pkg/runtime/serializer"
+	kubectrlmgrconfig "k8s.io/kubernetes/pkg/controller/apis/config"
+)
+
+// Funcs returns the fuzzer functions for the kube-controller manager apis.
+func Funcs(codecs runtimeserializer.CodecFactory) []interface{} {
+	return []interface{}{
+		func(obj *kubectrlmgrconfig.KubeControllerManagerConfiguration, c fuzz.Continue) {
+			c.FuzzNoCustom(obj)
+			// add GenericControllerManagerConfiguration values
+			obj.Generic.Address = fmt.Sprintf("%d.%d.%d.%d", c.Intn(256), c.Intn(256), c.Intn(256), c.Intn(256))
+			obj.Generic.MinResyncPeriod = metav1.Duration{Duration: time.Duration(c.Int63()) * time.Hour}
+			obj.Generic.ClientConnection.ContentType = "application/vnd.kubernetes.protobuf"
+			obj.Generic.ClientConnection.QPS = c.Float32()
+			obj.Generic.ClientConnection.Burst = c.Int31()
+			obj.Generic.ControllerStartInterval = metav1.Duration{Duration: time.Duration(c.Int63()) * time.Second}
+			obj.Generic.LeaderElection.LeaseDuration = metav1.Duration{Duration: time.Duration(c.Int63()) * time.Second}
+			obj.Generic.LeaderElection.RenewDeadline = metav1.Duration{Duration: time.Duration(c.Int63()) * time.Second}
+			obj.Generic.LeaderElection.RetryPeriod = metav1.Duration{Duration: time.Duration(c.Int63()) * time.Second}
+			obj.Generic.LeaderElection.ResourceLock = "endpoints"
+			obj.Generic.LeaderElection.LeaderElect = c.RandBool()
```

Comment:
aren't all these non-custom-format random values set by `c.FuzzNoCustom` above already?

Reply:
But in that case, https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/apis/config/v1alpha1/defaults.go#L51, we shoud set value like that `obj.TTLAfterFinishedController.ConcurrentTTLSyncs = c.Int31()`